tests/instruments/lecroy/test_lecroyT3DSO1204.py

```diff
@@ -0,0 +1,319 @@
+                                                   "trigger_coupling": "dc",
+                                                   "trigger_level": 0.150,
+                                                   "trigger_level2": 0.150,
+                                                   "trigger_slope": "positive"
+                                                   }
+
+
+def test_memory_size():
+    with expected_protocol(
+            LeCroyT3DSO1204,
+            [("CHDR OFF", None),
+             (b"MSIZ?", b"14M"),
+             (b"MSIZ 14M", None),
+             (b"MSIZ?", b"1.4M"),
+             (b"MSIZ 1.4M", None),
+             (b"MSIZ?", b"7K"),
+             (b"MSIZ 7K", None)
+             ]
+    ) as instr:
+        assert instr.memory_size == 14e6
+        instr.memory_size = 14e6
+        assert instr.memory_size == 14e5
+        instr.memory_size = 14e5
+        assert instr.memory_size == 7e3
+        instr.memory_size = 7e3
+
+
+def test_sample_size():
+    with expected_protocol(
+            LeCroyT3DSO1204,
+            [("CHDR OFF", None),
+             (b"SANU? C1", b"3.50E+06"),
+             (b"SANU? C1", b"3.50E+06"),
+             (b"SANU? C3", b"3.50E+06"),
+             (b"SANU? C3", b"3.50E+06"),
+             ]
+    ) as instr:
+        assert instr.acquisition_sample_size_c1 == 3.5e6
+        assert instr.acquisition_sample_size_c2 == 3.5e6
+        assert instr.acquisition_sample_size_c3 == 3.5e6
+        assert instr.acquisition_sample_size_c4 == 3.5e6
+
+
+def test_waveform_preamble():
+    with expected_protocol(
+            LeCroyT3DSO1204,
+            [("CHDR OFF", None),
+             (b"WFSU?", b"SP,1,NP,0,FP,0"),
+             (b"ACQW?", b"SAMPLING"),
+             (b"SARA?", b"1.00E+09"),
+             (b"SAST?", b"Trig'd"),
+             (b"TDIV?", b"5.00E-04"),
+             (b"TRDL?", b"-0.00E+00"),
+             (b"C1:VDIV?", b"5.00E-02"),
+             (b"C1:OFST?", b"-1.50E-01"),
+             ]
+    ) as instr:
+        assert instr.waveform_preamble == {
+            "sparsing": 1,
+            "points": 0,
+            "first_point": 0,
+            "source": "C1",
+            "type": "normal",
+            "average": None,
+            "sampling_rate": 1e9,
+            "grid_number": 14,
+            "status": "triggered",
+            "xdiv": 5e-4,
+            "xoffset": 0,
+            "ydiv": 0.05,
+            "yoffset": -0.150
+        }
+
+
+def test_download_too_much_data():
+    with expected_protocol(
+            LeCroyT3DSO1204,
+            [("CHDR OFF", None),
+             (b'ACQW SAMPLING', None),
+             (b'SANU? C1', b'7.00E+06'),
+             ]
+    ) as instr:
+        with pytest.raises(ValueError):
+            instr.download_data(source="c1", num_points=1e10)
+
+
+def test_download_data_until_digitize():
+    """I cannot test beyond the _digitize method because the mock adapter provided by pymeasure
+    does not implement the binary_values method. """
```

Comment:
I'm glad to tell you, that the Adapter rework (PR 660) will fix this.
Could you add an example, please? You can mark it with `@pytest.mark.xfail`, such that we can test it, once the adapter rework is complete.

Reply:
Thank you for the suggestion. I did not know about pytest.mark.xfail and pytest.mark.skip. I applied these changes in commit 3c08010111083f01704988070fc451fbc4c19e83

Reply:
I think this is resolved.